e2etest/GuestProxyAgentTest/Scripts/GuestProxyAgentExtensionValidation.ps1

```diff
@@ -107,32 +108,34 @@ Write-Output "TEST: ProxyAgent version running in VM is the same as expected ver
 $proxyAgentExeCmd = $extensionFolder + "\ProxyAgent\ProxyAgent\GuestProxyAgent.exe --version"
 $proxyAgentVersion = Invoke-Expression $proxyAgentExeCmd
 Write-Output "proxy agent version from extension folder: $proxyAgentVersion"
-$guestProxyAgentExtensionVersion = $false
+$guestProxyAgentExtensionVersion = $true
 $proxyAgentStatus = $json.status.substatus[1].formattedMessage.message
 $jsonObject = $proxyAgentStatus | ConvertFrom-json
 $extractedVersion = $jsonObject.version
-if ($extractedVersion -eq $proxyAgentVersion){ 
-    Write-Output "The proxy agent version matches the expected version"
-    $guestProxyAgentExtensionVersion = $true
-} else {
-    Write-Output "Error, the proxy agent version [ $extractedVersion ] does not match expected version [ $proxyAgentVersion ]"
+if ($expectedProxyAgentVersion -ne "0") {
+    if ($extractedVersion -eq $proxyAgentVersion -and $extractedVersion -eq $expectedProxyAgentVersion){ 
```

Comment:
same here, you need always compare "$extractedVersion -eq $expectedProxyAgentVersion"